src/ipyradiant/visualization/interactive_exploration.py

```diff
@@ -0,0 +1,309 @@
+
+
+class InteractiveViewer(W.VBox):
+    expand_button = trt.Instance(W.Button)
+    undo_button = trt.Instance(W.Button)
+    remove_temp_nodes_button = trt.Instance(W.Button)
+    cyto_graph = trt.Instance(ipycytoscape.CytoscapeWidget)
+    selected_node = trt.Instance(ipycytoscape.Node, allow_none=True)
+    rdf_graph = trt.Instance(rdflib.graph.Graph, allow_none=True)
+    cyto_style = trt.List(allow_none=True)
+    cyto_graph_layout = trt.Unicode(default_value="cola")
+
+    @trt.default("expand_button")
+    def _create_expand_button(self):
+        button = W.Button(
+            description="Expand Upon Selected Node",
+            layout=W.Layout(width="50%", height="40px"),
+        )
+        button.on_click(self.expand_button_clicked)
+        return button
+
+    @trt.default("undo_button")
+    def _create_undo_button(self):
+        button = W.Button(
+            description="Undo Last Expansion",
+            layout=W.Layout(width="25%", height="40px"),
+            disabled=True,
+        )
+        button.on_click(self.undo_expansion)
+        return button
+
+    @trt.default("remove_temp_nodes_button")
+    def _create_remove_temp_nodes_button(self):
+        button = W.Button(
+            description="Remove Temporary Nodes",
+            layout=W.Layout(width="25%", height="40px"),
+            disabled=False,
+        )
+        button.on_click(self.remove_temp_nodes)
+        return button
+
+    @trt.default("selected_node")
+    def _create_default_selected_node(self):
+        return None
+
+    @trt.default("cyto_style")
+    def _create_cyto_style(self):
+        return DEFAULT_CYTO_STYLE
+
+    @trt.default("rdf_graph")
+    def _create_rdf_graph(self):
+        return rdflib.Graph()
+
+    @trt.default("cyto_graph")
+    def _create_cyto_graph(self):
+        return ipycytoscape.CytoscapeWidget()
+
+    @trt.default("layout")
+    def _create_layout(self):
+        return W.Layout(width="80%")
+
+    @trt.observe("cyto_graph")
+    def update_cyto_graph(self, change):
+        self.cyto_graph.set_layout(name=self.cyto_graph_layout)
+        self.cyto_graph.set_style(self.cyto_style)
+        # on is a callback for cyto_graph instance (must be set on each instance)
+        self.cyto_graph.on("node", "click", self.log_node_clicks)
+        # Here we have to set the children again so that the changes propogate to the front end
+        # automatically. Ideally this would be done with traits but did not seem to work. LINK TO GITHUB ISSUE:
+        # https://github.com/jupyrdf/ipyradiant/issues/79
+        self.children = (
+            self.cyto_graph,
+            W.HBox(
+                children=[
+                    self.expand_button,
+                    self.undo_button,
+                    self.remove_temp_nodes_button,
+                ]
+            ),
+        )
+
+    @trt.validate("children")
+    def validate_children(self, proposal):
+        children = proposal.value
```

Comment:
Add a docstring that says this is necessary because `@trt.default` does not work on `children`

Reply:
Note: need to add this disclaimer across repo.